src/grisp_gpio.erl

```diff
@@ -19,21 +25,60 @@
 -export([code_change/3]).
 -export([terminate/2]).
 
+%--- Macros --------------------------------------------------------------------
+
+-define(EVENT_MSG_CODE,    $E).
+-define(CMD_START_POLLING,  1).
+-define(CMD_STOP_POLLING,   2).
+-define(CMD_PIN_CONFIG,    10).
+-define(CMD_PIN_GET,       11).
+-define(CMD_PIN_CLEAR,     12).
+-define(CMD_PIN_SET,       13).
+
+-define(EVENT_SERVER, grisp_gpio_events).
+
 %--- Records -------------------------------------------------------------------
 
--record(state, {driver}).
+-record(state, {
+    driver      :: {module(), reference()}
+}).
 
 %--- API -----------------------------------------------------------------------
 
 % @private
 start_link(DriverMod) ->
     gen_server:start_link({local, ?MODULE}, ?MODULE, DriverMod, []).
 
-configure(Pin, Type) -> configure(Pin, Type, [default]).
+configure(Pin, {Type, HTrigger, LTrigger, PollInterval, DebounceCounter, Attr}) ->
+    configure(Pin, Type, HTrigger, LTrigger, PollInterval, DebounceCounter, Attr);
+configure(Pin, {Type, HTrigger, LTrigger, PollInterval, Attr}) ->
+    configure(Pin, Type, HTrigger, LTrigger, PollInterval, Attr);
+configure(Pin, {Type, Attr}) ->
+    configure(Pin, Type, Attr);
+configure(Pin, Type) ->
+    configure(Pin, Type, [default]).
 
 configure(Pin, Type, Attr) ->
-    Command = <<(index(Pin)):8, 1:8, (map_type(Type)):8, (map_attr(Attr)):8>>,
-    gen_server:call(?MODULE, {command, Command}).
+    configure(Pin, Type, false, false, 1, 0, Attr).
+
+configure(Pin, Type, HTrigger, LTrigger, PollInterval, Attr) ->
+    configure(Pin, Type, HTrigger, LTrigger, PollInterval, 0, Attr).
+
+configure(Pin, Type, HTrigger, LTrigger, PollInterval, DebounceCounter, Attr)
```

Comment:
This API is very cumbersome. We should use an options map to configure the specifics instead of separate arguments.